crates/database/interface/src/lib.rs

```diff
@@ -52,7 +52,7 @@ pub trait DatabaseCommit {
 #[auto_impl(&, &mut, Box, Rc, Arc)]
 pub trait DatabaseRef {
     /// The database error type.
-    type Error;
+    type Error: core::error::Error;
```

Comment:
I would consider adding a `'static` bound here, as described in https://github.com/bluealloy/revm/pull/1840/files#r1819000225